### PR TITLE
Upgrade PHPStan to Level 8

### PR DESCRIPTION
**Pull request by joshbonnick**
This PR increases the PHPStan level to 8, fixing errors at each level.

## Notes
- Had to add `meta_image` as an `@property` in Rfc model as Larastan is unable to detect it when using raw DB query in migration.
- The level 6 commit is the majority of the changes as it included adding return types to almost every controller method.
- Level 8 added a few `@var` tags and `@property` tags.

### CODE REVIEW

File: app/Http/Livewire/ArgumentList.php [hunk 1]
```diff
@@ -28,11 +29,14 @@ class ArgumentList extends Component
 
     public ?Argument $showingComments = null;
 
+    /**
+     * @var string[]
```
Comment (Plytas) [inline app/Http/Livewire/ArgumentList.php:33]:
```suggestion
     * @var array<string, string>
```

File: app/Http/Livewire/RfcCounter.php [hunk 2]
```diff
@@ -12,18 +13,21 @@ class RfcCounter extends Component
 
     public Rfc $rfc;
 
+    /**
+     * @var string[]
```
Comment (Plytas) [inline app/Http/Livewire/RfcCounter.php:17]:
```suggestion
     * @var array<string, string>
```

File: app/Http/Livewire/VoteBar.php [hunk 3]
```diff
@@ -20,20 +21,23 @@ class VoteBar extends Component
 
     public ?string $body = null;
 
+    /**
+     * @var string[]
```
Comment (Plytas) [inline app/Http/Livewire/VoteBar.php:25]:
```suggestion
     * @var array<string, string>
```

File: app/Http/Middleware/AdminMiddleware.php [hunk 4]
```diff
@@ -3,11 +3,16 @@
...
 use Closure;
+use Illuminate\Http\RedirectResponse;
 use Illuminate\Http\Request;
+use Symfony\Component\HttpFoundation\Response;
 
 class AdminMiddleware
 {
-    public function handle(Request $request, Closure $next)
+    /**
+     *  @param  Closure(Request): (Response)  $next
```
Comment (Plytas) [inline app/Http/Middleware/AdminMiddleware.php:13]: Are parentheses around `Response` needed?

**Review by Plytas — CHANGES_REQUESTED** [review]

File: app/Http/Middleware/AdminMiddleware.php (hunk 4 above)
Comment (joshbonnick) [reply app/Http/Middleware/AdminMiddleware.php:13]: @Plytas Not 100% sure, it is from the 10.x skeleton 

https://github.com/laravel/laravel/blob/32ecad53a9a59d7ad33dbfd6352a5ea0fd0da2cf/app/Http/Middleware/RedirectIfAuthenticated.php#L16C103-L16C103

Comment (Plytas) [reply app/Http/Middleware/AdminMiddleware.php:13]: I'm pretty sure it's not needed.

Comment (joshbonnick) [reply app/Http/Middleware/AdminMiddleware.php]: It does not complain so we will go with no parentheses.

**Comment by joshbonnick** [comment]
> Have you looked into what it would take to go to level 9 and/or using [phpstan-strict-rules](https://github.com/phpstan/phpstan-strict-rules) and [Bleeding Edge](https://phpstan.org/blog/what-is-bleeding-edge)?

Not with this project, in other projects I have found that you start getting errors from the framework and packages that just lead to you writing code around PHPStan when the original was correct to start with.

Example, in `AppServiceProvider`, PHPStan does not know `$chromePath` is a string, so it needs to be cast or possibly requires an extra if statement.

This particular example is easily solved but it leads to unnecessarily verbose code throughout the project. 
```php
$this->app->singleton(Browsershot::class, function () {
    if (config('services.browsershot.fake')) {
        return new FakeBrowsershot();
    }

    $browsershot = new Browsershot();

    if ($chromePath = config('services.browsershot.chrome_path')) {
        $browsershot->setChromePath($chromePath);
    }

    // ...function continues
});
```

---

Another example, `$request->validated()` can return the entire validated request or be passed a key for a single field. We know the `validated` request is returning an array but to prevent the error we must swap to `$request->safe()->toArray()` or assign it to a temporary variable just to add `@var`
```php
public function store(RfcRequest $request): RedirectResponse
{
    $rfc = new Rfc($request->validated());
    $rfc->save();

    flash('Success');

    return redirect()->action([RfcEditController::class, 'edit'], $rfc);
}
```

There a total of 13 errors without bleeding edge & strict rules, all similar cases to the ones mentioned above.

---

If the maximum level of static analysis is what were are aiming for, then I can fix level 9 errors but I don't think the benefit of the strictness outweighs the drawbacks of writing code to that level of strictness.

**Comment by brendt** [comment]
I agree! It's useful as long as we don't have to write too much code just to keep PHPStan happy. Thanks a lot for this PR!

**Pull request merged**